sql/src/main/java/io/crate/expression/scalar/arithmetic/MapFunction.java

```diff
@@ -90,7 +78,17 @@ public final Object evaluate(TransactionContext txnCtx, Input<Object>... args) {
         return m;
     }
 
-    public static void register(ScalarFunctionModule scalarFunctionModule) {
-        scalarFunctionModule.register(NAME, RESOLVER);
+    public static void register(ScalarFunctionModule module) {
+        module.register(
+            Signature.builder()
+                .name(new FunctionName(null, NAME))
+                .kind(FunctionInfo.Type.SCALAR)
+                .typeVariableConstraints(List.of(typeVariableOfAnyType("V")))
+                .argumentTypes(parseTypeSignature("text"), parseTypeSignature("V"))
+                .returnType(parseTypeSignature("object(text, V)"))
```

Comment:
This is not really correct, depending on the argument types it will change to `object(text, V, text, V2, ...)` where `V` and `V2` may be bound to different types.
Currently the return type is not used so it won't matter, but will be later used for information of all registered functions (`pg_proc`).
Not sure, shall we define it as `object()` (undefined inner types)? We could also discuss/change it when adding the signature information to `pg_proc`.

Reply:
Not sure how to deal with this, but maybe meanwhile add a comment inline to point this out.